File: PYTHON/EASY/0217-contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Convert the input list 'nums' to a set, eliminating duplicates
        hashset = set(nums)

        # Check if the length of the set is different from the length of the original list
        # If there are duplicates, the set will have fewer elements than the list
        return len(hashset) != len(nums)

File: C#/EASY/0217-contains_duplicate.cs
public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        // Create a HashSet from the array 'nums'
        // A HashSet automatically removes duplicates, so the count will be less than the original array's length if there are duplicates
        return new HashSet<int>(nums).Count() != nums.Count();
    }
}
